File: bfs/bfs.py
graph = {
    'A' : ['B','C'],
    'B' : ['D', 'E', 'B'],
    'C' : ['F'],
    'D' : [],
    'E' : ['F'],
    'F' : ["G"],
    'G' : ['B'],
}

visited = set() # Set to keep track of visited nodes.
queue = set()


def bfs(visited, graph, node, target):

    visited.add(node)
    queue.add(node)
    
    while queue:
        node = queue.pop()

        if node == target:
            return True
            
        for neighbour in graph[node]:
            if neighbour not in visited:
                visited.add(neighbour)
                queue.add(neighbour)
                if neighbour == target:
                    return True
    return False


found = bfs(visited, graph, 'A', 'G')

print(found)

File: bfs/bfs.js
const graph = {
  A: ["B", "C"],
  B: ["D", "E"],
  C: ["F"],
  D: [],
  E: ["F"],
  F: ["G"],
  G: ["B"],
};

const visited = new Set();
const queue = [];

const bfs = (visited, graph, node, target) => {
  visited.add(node);
  queue.push(node);

  while (queue.length > 0) {
    const node = queue.shift();

    if (node === target) {
      return true;
    }

    const neighbours = graph[node];
    for (let i = 0; i < neighbours.length; i++) {
      if (!visited.has(neighbours[i])) {
        visited.add(neighbours[i]);
        queue.push(neighbours[i]);

        if (neighbours[i] === target) {
          return true;
        }
      }
    }
  }
};

const found = bfs(visited, graph, "A", "G");

console.log(found ? "yes" : "no");
